### PR TITLE
Add uri_encode (#273)

### PR DESCRIPTION
**Pull request by dr0i**
Use metamorph's URLEncoder for uri_encode.

Supersedes https://github.com/metafacture/metafacture-fix/pull/285.

### CODE REVIEW

File: metafix/src/main/java/org/metafacture/metafix/FixMethod.java [hunk 1]
```diff
@@ -662,8 +663,16 @@ public void apply(final Metafix metafix, final Record record, final List<String>
         public void apply(final Metafix metafix, final Record record, final List<String> params, final Map<String, String> options) {
             record.transform(params.get(0), s -> s.toUpperCase());
         }
+    },
+    uri_encode {
+        @Override
+        public void apply(final Metafix metafix, final Record record, final List<String> params, final Map<String, String> options) {
+            URL_ENCODER.setPlusForSpace(false);
```
Comment (blackwinter) [inline metafix/src/main/java/org/metafacture/metafix/FixMethod.java:670]: Preferrably, we would expose this as an option `plus_for_space` (along with `safe_chars`). But I don't think it's necessarily a blocker.

File: metafix/src/main/java/org/metafacture/metafix/FixMethod.java [hunk 2]
```diff
@@ -662,8 +663,16 @@ public void apply(final Metafix metafix, final Record record, final List<String>
...
+            record.transform(params.get(0), URL_ENCODER::process);
+        }
     };
 
+    private static final URLEncode URL_ENCODER = new URLEncode();
```
Comment (blackwinter) [inline metafix/src/main/java/org/metafacture/metafix/FixMethod.java:675]: There is no connection with `NAMED_GROUP_PATTERN`. Please move it to the end, separated by an empty line below `RANDOM`.

_If_ it should be a singleton at all, especially when we're making it configurable (see above).

**Review by blackwinter — CHANGES_REQUESTED** [review]

File: metafix/src/main/java/org/metafacture/metafix/FixMethod.java (hunk 2 above)
Comment (dr0i) [reply metafix/src/main/java/org/metafacture/metafix/FixMethod.java]: Good point. Moved instantiation into method in [c20c2a9](https://github.com/metafacture/metafacture-fix/pull/314/commits/c20c2a945d60fd3a52f3a218bc752b4561069a90).

File: metafix/src/main/java/org/metafacture/metafix/FixMethod.java (hunk 1 above)
Comment (dr0i) [reply metafix/src/main/java/org/metafacture/metafix/FixMethod.java]: Agreed - we can, if needed, easily enhance to use options.

**Comment by blackwinter** [comment]
Would you also document this new function in the README?

File: metafix/src/main/java/org/metafacture/metafix/FixMethod.java (hunk 1 above)
Comment (blackwinter) [reply metafix/src/main/java/org/metafacture/metafix/FixMethod.java]: Okay, but I'll unresolve it as a reminder.

**Comment by dr0i** [comment]
Updated the README.

**Comment by blackwinter** [comment]
Now we're entering shaky territory ;) We differentiate between "record-level" and "field-level" methods. You placed the implementation in the latter section, but the documentation in the former. I don't think we have clear criteria for which is which (@fsteeg, @TobiasNx?), but this one is almost definitely "field-level" I would say (so the documentation should match the implementation).

**Comment by dr0i** [comment]
No, makes sense. Wasn't aware of two sections.  Moved docu snippet  to the proper place. Please have a look again.

**Comment by blackwinter** [comment]
Sorry, but now it's in "script-level" ;) I've moved it down in db84fb7.

**Comment by dr0i** [comment]
Ahhh -  I put it into `Script-level functions` ... will fix that.

**Comment by blackwinter** [comment]
And I named the wrong section in the commit message :( Probably started the weekend a little early ;)

**Comment by dr0i** [comment]
>  Hope that's okay.

timely :)
Going to merge now.

**Pull request merged**